testsV2/connection_test.go

```diff
@@ -18,14 +18,13 @@ import (
 	"context"
 	"database/sql"
 	"testing"
+	"time"
 )
 
-// proxyConnTest is a test helper to verify the proxy works with a basic connectivity test.
-func proxyConnTest(t *testing.T, connName, driver, dsn string, port int, dir string) {
-	ctx := context.Background()
-
-	args := []string{connName}
+const connTestTimeout = time.Minute
 
+// proxyConnTest is a test helper to verify the proxy works with a basic connectivity test.
+func proxyConnTest(t *testing.T, ctx context.Context, args []string, driver, dsn string, port int, dir string) {
```

Comment:
Why did we move ctx/args out of this function into the callers? The point of this function is to avoid writing the same test 3 times, but now we've just done that with ctx.